### DISCUSSION TITLE
Is it possible to set math like this in typst?

### DISCUSSION THREAD

**Discussion by andrewragy** [typst/typst#2965] [Q&A]
![image](https://github.com/typst/typst/assets/39835161/988bf2e7-c2f0-44fa-ae48-8d33b548b989)

This equation is set using Times New Roman for the letters and digits and a math font for everything else. Is it possible to recreate it in Typst?

**Comment by xkevio** [comment]
Something like:
```typ
#show math.equation: eq => {
  show regex("\w"): set text(font: <your-font>)
  eq
}

$ integral(10x^5 + 6x^3 - 3/x^2) dif x $
```

should work. This does seem to break spacing for `dif` though.

**Reply by PNBRQK** [reply]
Since my question is similar to the OP's I wouldn't start a separate thread:

How to define the show rule for normal texts inserted in a math equation?

For example:
```typst
#set page(width: 12cm, height: auto, margin: (x:5pt, y: 10pt))
#set text(font: "Noto Sans Mono CJK SC", weight: "light", size: 12pt)

字体
$ 2+3 =^("123") 3+2 $
$ 2+3 =^("字体") 3+2 $    /* apparently "字体" here is not affected 
							by the previous #set text() */

#show math.equation: eq => {
	show regex("\w") /* or what should be in? */ : set text(font: "Noto Sans Mono CJK SC")
	eq
} 

$ 2+3 =^("字体") 3+2 $   /* but the font of 2+3 also has changed */
```  
![a](https://github.com/typst/typst/assets/9943990/60ffca4a-3584-439f-b933-5252a8bf35c1)

What I want is to just change the font face of normal text strings (especially of CJK chars, such as "字体") inside the math environment, and leave the rest as-is.
